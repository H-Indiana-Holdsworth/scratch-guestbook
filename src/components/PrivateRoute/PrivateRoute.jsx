import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { useUser } from '../../hooks/useUser';

export function PrivateRoute({ children, ...routeProps }) {
  let user = useUser();

  return (
    <Route
      {...routeProps}
      render={({ location }) =>
        user.user ? children : <Redirect to={{ pathname: '/auth', state: { from: location } }} />
      }
    />
  );
}
