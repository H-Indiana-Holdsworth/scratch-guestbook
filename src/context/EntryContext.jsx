import { useContext, createContext, useState } from 'react';

const EntriesContext = createContext();

function EntriesProvider({ children }) {
  const [entries, setEntries] = useState([]);

  return (
    <EntriesContext.Provider value={{ entries, setEntries }}> {children} </EntriesContext.Provider>
  );
}

const useEntries = () => {
  const context = useContext(EntriesContext);

  if (context === undefined) {
    throw new Error('useEntries must be defined within a UserContext Provider');
  }

  return context;
};

export { EntriesProvider, useEntries };
