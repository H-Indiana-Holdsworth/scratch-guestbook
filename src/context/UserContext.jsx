import { createContext, useState } from 'react';

export const UserContext = createContext();

export function UserProvider({ children }) {
  const [user, setUser] = useState('');

  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.REACT_APP_AUTH_EMAIL &&
      password === process.env.REACT_APP_AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (e) => {
    e.preventDefault();
    setUser(null);
  };

  return <UserContext.Provider value={{ user, login, logout }}>{children}</UserContext.Provider>;
}
