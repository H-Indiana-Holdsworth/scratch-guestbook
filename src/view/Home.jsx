import React from 'react';
import Entry from '../components/Entry/Entry';
import EntryList from '../components/EntryList/EntryList';
import GuestForm from '../components/GuestForm/GuestForm';
import { EntriesProvider } from '../context/EntryContext';

export default function Home() {
  return (
    <div>
      <EntriesProvider>
        <EntryList>
          <Entry />
        </EntryList>
      </EntriesProvider>
      <GuestForm />
    </div>
  );
}
