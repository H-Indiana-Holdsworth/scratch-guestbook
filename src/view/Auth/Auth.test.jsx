import { Switch } from '@material-ui/core';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { PrivateRoute } from '../../components/PrivateRoute/PrivateRoute';
import { UserProvider } from '../../context/UserContext';
import Home from '../Home';
import Auth from './Auth';

test('Should render the login view', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Switch>
          <Route exact path="/auth">
            <Auth />
          </Route>
          <PrivateRoute exact path="/">
            <Home />
          </PrivateRoute>
        </Switch>
      </MemoryRouter>
    </UserProvider>
  );

  expect(container).toMatchSnapshot();
});
