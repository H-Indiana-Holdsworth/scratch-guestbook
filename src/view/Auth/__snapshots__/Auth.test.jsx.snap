// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the login view 1`] = `
<div>
  <span
    class="MuiSwitch-root"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4 MuiSwitch-input"
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
</div>
`;
